### PR TITLE
feature: Add models_v2 under lineage context

### PR DESCRIPTION
**Pull request by yzhu0**
*Issue #, if available:*

*Description of changes:*
Add models_v2 under lineage context. models_v2 return list of models by using lineage query

*Testing done:*
- unit test:
`tox -e py36 -- tests/unit/sagemaker/lineage/test_endpoint_context.py -vv`

tests/unit/sagemaker/lineage/test_endpoint_context.py::test_models PASSED                                                                                                                                                              [ 50%]
tests/unit/sagemaker/lineage/test_endpoint_context.py::test_models_v2 PASSED                                                                                                                                                           [100%]
==========================================================================================2 passed in 2.04s ========================================================================================

- integ test:
`tox -e py36 -- tests/integ/sagemaker/lineage/test_endpoint_context.py`

collected 5 items
tests/integ/sagemaker/lineage/test_endpoint_context.py::test_model PASSED                                                                                                                                                              [ 50%]
tests/integ/sagemaker/lineage/test_endpoint_context.py::test_model_v2 PASSED                                                                                                                                                           [100%]
.....
tests/integ/sagemaker/lineage/test_endpoint_context.py .....
========================================================================================= 5 passed in 14.45s =========================================================================================
## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integ/sagemaker/lineage/test_endpoint_context.py [hunk 1]
```diff
@@ -24,11 +27,22 @@ def test_model(
...
     for model in model_list:
         assert model.source_arn == endpoint_action_obj.action_arn
-        assert model.destination_arn == model_obj.context_arn
+        assert model.destination_arn == model_obj.artifact_arn
         assert model.source_type == "ModelDeployment"
         assert model.destination_type == "Model"
 
 
+def test_model_v2(endpoint_context_associate_with_model, model_obj, sagemaker_session):
+    time.sleep(SLEEP_TIME_ONE_SECONDS)
```
Comment (staubhp) [inline tests/integ/sagemaker/lineage/test_endpoint_context.py:36]: What's this for?

Comment (yzhu0) [reply tests/integ/sagemaker/lineage/test_endpoint_context.py:36]: Looks like the lineage ingestion take a while, sometimes I get 0 length for the query lineage. So I add 1 second sleep here.

Comment (yzhu0) [reply tests/integ/sagemaker/lineage/test_endpoint_context.py:36]: > 

Looks like the lineage ingestion take a while, sometimes I get 0 length for the query lineage. So I add 1 second sleep here.

**Review by staubhp — DISMISSED** [review]

**Review by staubhp — APPROVED** [review]

**Review by ahsan-z-khan — DISMISSED** [review]

File: src/sagemaker/lineage/context.py [hunk 2]
```diff
@@ -283,6 +283,46 @@ def models(self) -> List[association.Association]:
...
+        query_filter = LineageFilter(
+            entities=[LineageEntityEnum.ACTION], sources=[LineageSourceEnum.MODEL_DEPLOYMENT]
+        )
+        model_deployment_query_result = LineageQuery(self.sagemaker_session).query(
+            start_arns=[self.context_arn],
+            query_filter=query_filter,
+            direction=direction,
+            include_edges=False,
+        )
+        model_deployment_vertices: [] = model_deployment_query_result.vertices
```
Comment (mufaddal-rohawala) [inline src/sagemaker/lineage/context.py:304]: Can `model_deployment_query_result.vertices` be None?
```
>>> m = [1,2,3]
>>> n = None
>>> o: [] = m
>>> o
[1, 2, 3]
>>> o: [] = n
>>> print(o)
None
```

File: src/sagemaker/lineage/context.py [hunk 3]
```diff
@@ -283,6 +283,46 @@ def models(self) -> List[association.Association]:
         ]
         return model_list
 
+    def models_v2(
+        self, direction: LineageQueryDirectionEnum = LineageQueryDirectionEnum.DESCENDANTS
```
[the remark this answers is not in the document]
Comment (yzhu0) [reply src/sagemaker/lineage/context.py:287]: > Please add args docstrng.

added docstring

File: src/sagemaker/lineage/context.py (hunk 2 above)
Comment (yzhu0) [reply src/sagemaker/lineage/context.py]: > Can `model_deployment_query_result.vertices` be None?
> 
> ```
> >>> m = [1,2,3]
> >>> n = None
> >>> o: [] = m
> >>> o
> [1, 2, 3]
> >>> o: [] = n
> >>> print(o)
> None
> ```

Fixed

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by ahsan-z-khan — APPROVED** [review]

**Pull request merged**